### DISCUSSION TITLE
Search for key in dictionaries of documents

### DISCUSSION THREAD

**Discussion by pippopazzo** [ravendb/ravendb#14237] [Q&A]
The following code does not throw any exception, but does not find any document (even when it should):

```
var items = session.Query<ObjectWithDictionary>()
    .Search(x => x.MyDictionary.Keys, id)
    .ToListAsync();
```

On the other hand, if I avoid dictionaries and use, for example, a list of nested objects containing a string, as below, I get a runtime exception:

```
var items = session.Query<ObjectWithListOfNestedObjects>()
    .Search(x => x.MyNestedObjects.Select(x => x.Id), id)
    .ToListAsync();
```

Is it only allowed to search in a `ICollection<string>`, like in the "Hobbies" example that you provide?

Many thanks!

**Comment by garayx** [comment]
Hi

Can you share a [unit-test](https://ravendb.net/docs/article-page/5.3/Csharp/start/test-driver) or sample app that demonstrate the issue?

**Comment by ayende** [comment]
I'm fairly certain that the linq provider will not support this as is. The easiest way to handle that is to use a static index.
